### PR TITLE
Matrix trace

### LINKED ISSUE #34: Add matrix trace

### PR DESCRIPTION
**Pull request by Aertalon**

### CODE REVIEW

File: src/matrix_ops.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+inline constexpr auto identity = detail::make_identity<V>();
+
+// FIXME: this should be a matrix method
+template <class V, std::size_t Cols>
+requires(matrix<V, Cols>::is_square)
+    [[nodiscard]] constexpr auto trace(matrix<V, Cols> const& m) -> scalar_t<V>
+{
+    return [&m]<std::size_t... Is>(std::index_sequence<Is...>)
+    {
+        return (m.template get<Is, Is>() + ...);
```
Comment (oliverlee) [inline src/matrix_ops.hpp:32]: nice :)

**Review by oliverlee — APPROVED** [review]

File: src/matrix_ops.hpp (hunk 1 above)
Comment (Aertalon) [reply src/matrix_ops.hpp:32]: Is there really no way to instruct the compiler to consider get a template method? One of the last commits in the chain is basically unreadable due to the all .template things :/

**Comment by oliverlee** [comment]
We can implement pipelining 🙂

I wouldn't mind doing it sometime this week.

**Comment by oliverlee** [comment]
But you can also use for loops or create that cartesian product range
factory that's going into C++23.

**Comment by Aertalon** [comment]
Cartesian product would be quite useful to simplify the transpose and cycle over all elements (although that needs to be done with a view)

**Comment by Aertalon** [comment]
And yeah, maybe here a for loop would have been easier. Or a collection of iterators of some type that can be specialized to go over columns, OR over rows, or some crazy combination (in this case, over the diagonal elements)?

**Pull request merged**